### PR TITLE
Document current solr swap process.

### PR DESCRIPTION
**Pull request by dkinzer**

### CODE REVIEW

File: doc/catalog-search-oai-harvest-process.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+===
+
+## Summary
+The current catalog OAI harvesting process is broken up into two distinct but linked processes and their respective DAGs:
+* [The Pre Production Catalog OAI Harvest](../cob_datapipeline/prod_catalog_oai_harvest_dag.py)
+* [The Production Catalog OAI Harvest](../cob_datapipeline/catalog_preproduction_oai_harvest_dag.py)
+
+Each of these DAGs will harvest independently to a configured collection:
+* `catalog_pre_production_oai_harvest`  harvests:
+    * from date set in `CATALOG_PRE_PRODUCTION_SOLR_COLLECTION` variable.
```
Comment (bibliotechy) [inline doc/catalog-search-oai-harvest-process.md:11]: Is this the right variable name?

File: doc/catalog-search-oai-harvest-process.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    * until `Now()` by default.
+    * into the collection set in `CATALOG_PRE_PRODUCTION_HARVEST_FROM_DATE` variable.
+
+* `catalog_preproduction_oai_harvest_dag` harvests:
+    * from date set in `CATALOG_PROD_HARVEST_FROM_DATE` variable.
+    * until `Now()` by default.
+    * into the collection set in `CATALOG_PRODUCTION_SOLR_COLLECTION` variable
+
+## Collection swap process.
+When tul_cob gets deployed, the `CATALOG_PRODUCTION_SOLR_COLLECTION` will get
```
Comment (bibliotechy) [inline doc/catalog-search-oai-harvest-process.md:21]: You should be more specific: this only happens when a tul_cob release is deployed to production.

File: doc/catalog-search-oai-harvest-process.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+[automatically](https://github.com/tulibraries/tul_cob/blob/main/.circleci/update-airflow.sh)
+updated to what is deployed.
+
+The following steps MUST be taken when swaping/updating the production collection on tul_cob.
+* Before deploying to production we MUST stop the oai pre production and production dags and kill any running harvest.
+* After the deployment is complete we MUST update the
+  `CATALOG_PRODUCTION_SOLR_COLLECTION` variable with the
```
Comment (bibliotechy) [inline doc/catalog-search-oai-harvest-process.md:28]: I thought this was done automatically by the cod_prod deploy as you mention on line 21.

Comment (dkinzer) [reply doc/catalog-search-oai-harvest-process.md:28]: oops ... meant DATE. thanks for catching that.

File: doc/catalog-search-oai-harvest-process.md (hunk 1 above)
Comment (dkinzer) [reply doc/catalog-search-oai-harvest-process.md:11]: I believe so
image

**Comment by dkinzer** [comment]
> A few questions inline plus a more general one: Why is this a separate document from [full-reindex-process.md](https://github.com/tulibraries/cob_datapipeline/blob/master/doc/full-reindex-process.md)? That document covers up to the point of deploying a new release to production, so this feels like a subset of that, though more fleshed out in some places. Perhaps the full reindex one needs to be broekn up? Just want to avoid having them same info in multiple places.

I don't see this a a subset of the full reindex process like you do.  In fact the full reindex for this was completed many days ago... There is a disjoint there in the process.  In fact we can run the full reindex process without ever intending it to be used for a swap process.

File: doc/catalog-search-oai-harvest-process.md (hunk 1 above)
Comment (bibliotechy) [reply doc/catalog-search-oai-harvest-process.md:11]: So I read this as stating that we are harvesting from a date that is extracted from the datestamp part of the collection name variable. I don't think that is correct.
Shouldn't it be that we are harvesting from  date set in `CATALOG_PRE_PRODUCTION_HARVEST_FROM_DATE`?

Comment (dkinzer) [reply doc/catalog-search-oai-harvest-process.md:11]: Yeah, you're right. Copy pasta mistake.  Correcting.

**Comment by bibliotechy** [comment]
> I don't see this a a subset of the full reindex process like you do. In fact the full reindex for this was completed many days ago... There is a disjoint there in the process. In fact we can run the full reindex process without ever intending it to be used for a swap process.

You're point about this not being a subset of the full reindex process is well taken. I think my using the term "full reindex" to refer to the end-to-end Alma export -> release to production, is causing the issue. I think "Prod Solr Collection Swap Process" would be more useful description.

I do think there is value in documenting the full end to end swap process as well as each part of the process, but I want to avoid documentation drift between those two parts, especially about steps you MUST take in order to complete this successfully.

I'd like to propose that we 
1. rename [Full Reindex Process Steps](/tulibraries/cob_datapipeline/blob/main/doc/full-reindex-process.md) to Prod Solr Collection Swap Process
1. Break up the Prod Solr Collection Swap Process document into three sections
   1. Full Reindex
   1. OAI Harvests
   1. Deployment to Prod
1. Move the part of this PR from `### Production Collection swap process.` and below into the Prod Solr Collection Swap Process document in the Deployment to Prod section
1. Link to the appropriate sections of the Prod Solr Collection Swap Process document from this document.

What do you think?

**Comment by dkinzer** [comment]
Yes, that makes sense to me.

**Comment by bibliotechy** [comment]
@dkinzer OK, I'll make a PR to this branch

**Pull request merged**